File: Otros semestres/estructura de datos/Listas_Enlazada_Cicurlar/Lista_Circular.java
package Listas_Enlazada_Cicurlar_java;

public class Lista_Circular {
    Nodo Inicio;
    Nodo Fin;
    int size;

    public Lista_Circular(){
        Inicio = null;
        Fin = null;
        size = 0;
    }

    public void Addlista(int d){
        Nodo nuevNodo = new Nodo(d);
        if (Inicio == null) {
            Inicio = nuevNodo;
            Fin = nuevNodo;
            Fin.Siguiente = Inicio;
        }else{
            Fin.Siguiente = nuevNodo;
            Fin = nuevNodo;
            Fin.Siguiente = Inicio;
        }
        size++;
    }

    public int Size(){
        return size;
    }

    public void MostrarLista(){
        Nodo Temp = Inicio;
        if (Temp != null) {
            do {
                System.out.print(Temp.getDato() + " ");
                Temp = Temp.Siguiente;
            } while(Temp != Inicio);
            System.out.println();
        }
    }
}

File: Otros semestres/estructura de datos/Listas_Enlazada_Cicurlar/ListaCircular.cpp
#include<iostream>
using namespace std;

template <typename T>
class Nodo{

  private:
    T Dato;
    Nodo* Enlace;

  public:

    Nodo(T dato){
      Dato = dato;
      Enlace = nullptr;
    }
    T getDato(){
      return Dato;
    }
    Nodo* getEnlace(){
      return Enlace;
    }
    void setEnlace(Nodo* enlace){
      Enlace = enlace;
    }
};

template <typename U>
class ListaCircular{
private:
    Nodo<U>* Inicio;
    Nodo<U>* Final;

public:
    ListaCircular(){
        Inicio = nullptr;
        Final = nullptr;
    }

    void insertarFinal(U dato){
        Nodo<U>* nuevo = new Nodo<U>(dato);
        if (Inicio == nullptr){
            Inicio = nuevo;
            Final = nuevo;
            Final->setEnlace(Inicio);
        } else {
            Final->setEnlace(nuevo);
            Final = nuevo;
            Final->setEnlace(Inicio);
        }
    }

    void mostrarLista(){
        if (Inicio == nullptr) {
            cout << "La lista está vacía" << endl;
            return;
        }

        Nodo<U>* aux = Inicio;
        do {
            cout << aux->getDato() << endl;
            aux = aux->getEnlace();
        } while (aux != Inicio);
    }
};

int main(){
    ListaCircular<int>* lista = new ListaCircular<int>();
    lista->insertarFinal(5);
    lista->insertarFinal(6);
    lista->insertarFinal(7);
    lista->insertarFinal(8);
    lista->mostrarLista();

    return 0;
}